src/ashley/models.py

```diff
@@ -197,3 +204,45 @@ class Meta:
 
 # Default implementation of the AbstractLTIContext model
 LTIContext = model_factory(AbstractLTIContext)
+
+
+def image_directory_path(instance, filename):
+    """File will be uploaded to specific path"""
+    return "{0}_{1}/{2}/{3}".format(
+        instance.forum.id, instance.forum.slug, instance.poster.id, filename
+    )
+
+
+class AbstractUploadImage(DatedModel):
+    """Abstract model for images."""
+
+    file = models.ImageField(
+        upload_to=image_directory_path,
+        max_length=100,
+        blank=True,
+        validators=[
+            validate_upload_image_file_size,
+            validate_upload_image_allowed_type,
+        ],
```

Comment:
I'd use [FileExtensionValidator](https://docs.djangoproject.com/en/dev/ref/validators/#fileextensionvalidator) instead of the `validate_upload_image_allowed_type` validator.